File: Scanner/Models/Report.cs
using Newtonsoft.Json;
using Scanner.Reports.Types;

namespace Scanner.Models;

public class Report
{
    [JsonProperty("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required string DomainName { get; set; }
    public string? Error { get; set; }

    public string[] NameServers { get; set; } = [];
    public Address[] ApexAddresses { get; set; } = [];
    public string[] ApexText { get; set; } = [];
    
    public DateTime RequestedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
    public ReportStatus Status { get; set; } = ReportStatus.Created;
}

public enum ReportStatus
{
    Created,
    Processing,
    Failed,
    Completed
}

File: src/Frontend/src/data/report.ts
export interface DomainReport {
    id: string;
    domainName: string;
    error?: string;

    nameServers: NameServer[];
    apexAddresses: Address[];
    apexText: string[];

    requestedAt: string;
    completedAt?: string;
    status: ReportStatus;
}

export enum ReportStatus {
    Created = 0,
    Processing = 1,
    Failed = 2,
    Completed = 3
}

export interface Address {
    value: string;
    reverseName: string;
    asn: string;

    orgName: string;
    city: string;
    region: string;
    country: string;
}

export interface NameServer {
    hostname: string;
    serviceName: string;
}
